### PR TITLE
Planning: init open_space_trajectory_optimizer

### PR DESCRIPTION
**Pull request by JasonZhou404**

### CODE REVIEW

File: modules/planning/proto/open_space_trajectory_provider_config.proto [hunk 1]
```diff
@@ -3,4 +3,94 @@ syntax = "proto2";
...
+  optional double traj_back_penalty = 6 [default = 0.0];
+  optional double traj_gear_switch_penalty = 7 [default = 10.0];
+  optional double traj_steer_penalty = 8 [default = 100.0];
+  optional double traj_steer_change_penalty = 9 [default = 10.0];
+  optional double grid_a_star_xy_resolution = 15 [default = 0.1];
+  optional double node_radius = 16 [default = 0.5];
+}
+
+enum DualVariableMode {
+    DUAL_VARIABLE_IPOPT = 0;
```
Comment (xiaoxq) [inline modules/planning/proto/open_space_trajectory_provider_config.proto]: Consistent indents.

File: modules/planning/tasks/optimizers/open_space_trajectory_generation/open_space_trajectory_optimizer.h [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  void LoadTrajectory(const Eigen::MatrixXd& state_result_ds,
+                      const Eigen::MatrixXd& control_result_ds,
+                      const Eigen::MatrixXd& time_result_ds);
+
+  void Stop();
+
+ private:
+  bool IsInitPointNearDestination(
+      const common::TrajectoryPoint& planning_init_point,
+      const std::vector<double>& end_pose, const double& rotate_angle,
```
Comment (xiaoxq) [inline modules/planning/tasks/optimizers/open_space_trajectory_generation/open_space_trajectory_optimizer.h]: No need to apply "&" on primitive types.

File: modules/planning/tasks/optimizers/open_space_trajectory_generation/open_space_trajectory_optimizer_test.cc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+namespace planning {
+class OpenSpaceTrajectoryOptimizerTest : public ::testing::Test {
+ protected:
+  OpenSpaceTrajectoryOptimizerConfig config;
+};
+
+// dumy test
+TEST_F(OpenSpaceTrajectoryOptimizerTest, Init) {
+  OpenSpaceTrajectoryOptimizer open_space_trajectory_optimizer(config);
+  EXPECT_NE(&open_space_trajectory_optimizer, nullptr);
```
Comment (xiaoxq) [inline modules/planning/tasks/optimizers/open_space_trajectory_generation/open_space_trajectory_optimizer_test.cc]: Too dummy. Just remove.

File: modules/planning/tasks/optimizers/open_space_trajectory_generation/open_space_trajectory_optimizer_test.cc [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+#include "modules/planning/proto/open_space_trajectory_provider_config.pb.h"
+
+namespace apollo {
+namespace planning {
+class OpenSpaceTrajectoryOptimizerTest : public ::testing::Test {
+ protected:
+  OpenSpaceTrajectoryOptimizerConfig config;
+};
+
+// dumy test
```
Comment (xiaoxq) [inline modules/planning/tasks/optimizers/open_space_trajectory_generation/open_space_trajectory_optimizer_test.cc]: Dummy

File: modules/planning/proto/open_space_trajectory_provider_config.proto (hunk 1 above)
Comment (JasonZhou404) [reply modules/planning/proto/open_space_trajectory_provider_config.proto]: fixed

File: modules/planning/tasks/optimizers/open_space_trajectory_generation/open_space_trajectory_optimizer.h (hunk 2 above)
Comment (JasonZhou404) [reply modules/planning/tasks/optimizers/open_space_trajectory_generation/open_space_trajectory_optimizer.h]: fixed

File: modules/planning/tasks/optimizers/open_space_trajectory_generation/open_space_trajectory_optimizer_test.cc (hunk 3 above)
Comment (JasonZhou404) [reply modules/planning/tasks/optimizers/open_space_trajectory_generation/open_space_trajectory_optimizer_test.cc]: fixed

File: modules/planning/tasks/optimizers/open_space_trajectory_generation/open_space_trajectory_optimizer_test.cc (hunk 4 above)
Comment (JasonZhou404) [reply modules/planning/tasks/optimizers/open_space_trajectory_generation/open_space_trajectory_optimizer_test.cc]: fixed

**Pull request merged**